src/mds/MDBalancer.cc

```diff
@@ -579,6 +579,12 @@ void MDBalancer::queue_split(const CDir *dir, bool fast)
       return;
     }
 
+    if (!mds->is_active()) {
```

Comment:
```suggestion
    if (mds->is_stopping()) {
```

Reply:
@batrick I've intentionally made this !active, cause otherwise we'd need to test for `is_stopping` and `is_stopped` at least. Also, what's the point in proceeding in all the other transient and not-so-active states?

Reply:
> @batrick I've intentionally made this !active, cause otherwise we'd need to test for `is_stopping` and `is_stopped` at least. Also, what's the point in proceeding in all the other transient and not-so-active states?

`is_stopped()` is never true because an MDS respawns as soon as it's removed from the map.

My suggestion to use a narrow `is_stopping` check is to avoid potential regressions due to unknown coupling with other parts of the MDS (especially multiple ranks).

Reply:
OK that's a valid point, thanks. Changed.